WcaOnRails/spec/controllers/teams_controller_spec.rb

```diff
@@ -0,0 +1,148 @@
+      end
+    end
+
+    context "when signed in as admin" do
+      sign_in { FactoryGirl.create :admin }
+
+      it 'shows the teams index page' do
+        get :index
+        expect(response).to render_template :index
+      end
+    end
+
+    context 'when signed in as a regular user' do
+      sign_in { FactoryGirl.create :user }
+
+      it 'does not allow access' do
+        get :index
+        expect(response).to redirect_to root_url
+      end
+    end
+  end
+
+  describe "GET #new" do
+    context "when not signed in" do
+      sign_out
+
+      it 'redirects to the sign in page' do
+        get :new
+        expect(response).to redirect_to new_user_session_path
+      end
+    end
+
+    context "when signed in as admin" do
+      sign_in { FactoryGirl.create :admin }
+
+      it 'shows the teams index page' do
+        get :new
+        expect(response).to render_template :new
+      end
+    end
+
+    context 'when signed in as a regular user' do
+      sign_in { FactoryGirl.create :user }
+
+      it 'does not allow access' do
+        get :new
+        expect(response).to redirect_to root_url
+      end
+    end
+  end
+
+  describe 'POST #create' do
+    context 'when not signed in' do
+      it 'redirects to the sign in page' do
+        post :create, team: { name: "Team2016" }
+        expect(response).to redirect_to new_user_session_path
+      end
+    end
+
+    context 'when signed in as a regular user' do
+      sign_in { FactoryGirl.create :user }
+      it 'does not allow creation' do
+        post :create, team: { name: "Team2016" }
+        expect(response).to redirect_to root_url
+      end
+    end
+
+    context 'when signed in as an admin' do
+      sign_in { FactoryGirl.create :admin }
+
+      it 'creates a new team' do
+        post :create, team: { name: "Team2016" }
+        new_team = assigns(:team)
+        expect(response).to redirect_to edit_team_path(new_team)        
+        expect(new_team.name).to eq "Team2016"
+      end
+    end
+  end
+
+  describe 'POST #update' do
+    context 'when signed in as an admin' do
+      let(:admin) { FactoryGirl.create :admin }
+      before :each do
+        sign_in admin 
+      end
+
+      it 'can change name' do
+        patch :update, id: team, team: { name: "Hello" }
+        expect(response).to redirect_to edit_team_path(team)
+        new_team = assigns(:team)
+        expect(new_team.name).to eq "Hello"
+      end
+
+      it 'can change description' do
+        patch :update, id: team, team: { description: "This team is the best!" }
+        expect(response).to redirect_to edit_team_path(team)
+        new_team = assigns(:team)
+        expect(new_team.description).to eq "This team is the best!"
+      end
+
+      it 'can change friendly ID' do
+        patch :update, id: team, team: { friendly_id: "bestteam" }
+        expect(response).to redirect_to edit_team_path(team)
+        new_team = assigns(:team)
```

Comment:
Again, better to go straight to the database here. Just reload team and expect away.